pkg/build/builder/daemonless.go

```diff
@@ -305,32 +309,101 @@ func buildDaemonlessImage(sc types.SystemContext, store storage.Store, isolation
 
 func generateTransientMounts() []string {
 	mounts := []string{}
-	mounts = appendRHSMMount(mounts)
+	mounts = appendRHSMMount(defaultMountStart, mounts)
+	mounts = appendETCPKIMount(defaultMountStart, mounts)
+	mounts = appendRHRepoMount(defaultMountStart, mounts)
 	mounts = appendCATrustMount(mounts)
 	return mounts
 }
 
-func appendRHSMMount(mounts []string) []string {
-	st, err := os.Stat("/run/secrets/rhsm")
+func appendRHRepoMount(pathStart string, mounts []string) []string {
```

Comment:
I think you should remove the `pathStart string` from the parameters list and just use the `nodeCredentialsFile` constant within the functions, unless you think there is a good chance of someone wanting to change the path passed to the function in the future, taking into consideration that these functions are private to this package.

Reply:
that would break what I do with the unit tests @coreydaley 

when the real code runs, pathStart == defaultMountStart == /run/secrets
but we cannot manipulate /run/secrets in the unit tests so pathStart is a tmp dir the unit tests create and then clean up

also, you mean `defaultMountStart` not `nodeCredentialsFile`

Reply:
ok.  Yes, that is what i meant.